colabs/Wczytanie_danych.ipynb

```diff
@@ -843,7 +833,164 @@
+      "cell_type": "markdown",
+      "source": [
+        "## Deifniujemy metodę liczącą podobieństwo dwóch aktorów"
+      ],
+      "metadata": {
+        "id": "FhQGzHfgl5mF"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def similarity(actor_1_id, actor_2_id):\n",
+        "  # tworzymy słownik i zmienną similarity\n",
+        "  actors_dict = [{'actor_id': actor_1_id}, {'actor_id': actor_2_id}]\n",
+        "  similarity = 0\n",
+        "\n",
+        "  # załadowanie danych o aktorach do odpowiednich tabel\n",
+        "  data_1_2 = data.filter((data.nconst == actor_1_id) | (data.nconst == actor_2_id))\n",
+        "  actor_1_data = data.filter(data_1_2.nconst == actor_1_id)\n",
+        "  actor_2_data = data.filter(data_1_2.nconst == actor_2_id)\n",
+        "  actor_firsts = [actor_1_data.first(), actor_2_data.first()]\n",
+        "  \n",
+        "  # załadowanie danych do odpowiednich słowników\n",
+        "  for i in [0, 1]:    \n",
+        "    actors_dict[i]['knownForTiles'] = actor_firsts[i].knownForTitles.split(\",\")\n",
+        "    # gdyby brakowało informacji o dacie urodzenia\n",
+        "    try:\n",
+        "      bYear = int(actor_firsts[i].birthYear)\n",
+        "      dYear = int(actor_firsts[i].deathYear)\n",
+        "      # TO DO usunąć funkcję int(), gdy typ danych w kolumnach birthYear i deathYear zostanie zmieniony na poprawny\n",
+        "      # trzeba będzie wtedy też zmienić metodę radzenia sobie z brakującymi wartościami\n",
+        "    except ValueError:\n",
+        "      bYear = 0\n",
+        "      dYear = 0\n",
+        "    actors_dict[i]['yearsActivity'] = [bYear, dYear]\n",
+        "    \n",
+        "  actors_dict[0]['titles_id'] = actor_1_data.select('tconst').rdd.flatMap(lambda x: x).collect()\n",
+        "  actors_dict[1]['titles_id'] = actor_2_data.select('tconst').rdd.flatMap(lambda x: x).collect()\n",
+        "  \n",
+        "  # similairty na podstawie występowania w tych samych tytułach\n",
+        "  for title in actors_dict[0]['titles_id']:\n",
+        "    if title in actors_dict[1]['titles_id']:\n",
+        "      similarity += 0.25\n",
+        "  \n",
+        "  # similairty na podstawie bycia znanym z tych samych tytułów\n",
+        "  for title in actors_dict[0]['knownForTiles']:\n",
+        "    if title in actors_dict[1]['knownForTiles']:\n",
+        "      similarity += 3\n",
+        "\n",
+        "  # similairty na podstawie życia w tym samym okresie czasu\n",
+        "  years_overlap = max(0, min(actors_dict[0]['yearsActivity'][1], actors_dict[1]['yearsActivity'][1])- max(actors_dict[0]['yearsActivity'][0], actors_dict[1]['yearsActivity'][0]))\n",
+        "  similarity += years_overlap * 0.02\n",
+        "  # TO DO zoptymalizować przechodzenie przez listę lat\n",
+        "  \n",
+        "  return similarity\n",
+        "  # TO DO zestandaryzować wartośc zwracaną przez metodę do wartości z przedziału [-1, 1]"
```

Comment:
Ta funkcja nie da rady wytrzymać obliczeń. Na całym zbiorze. Możemy zostawić w mainie, ale tylko po to by zaraz to naprawić. 

Aby dobrze to zadziałało trzeba by mapować każdą kolumnę istotną na liczbę i zastosować operację wektorową.

Reply:
Dać mi znać co z tym chcemy zrobić. Mergować czy poprawiać?

Reply:
Na razie możemy to zostawić bez mergowania i będę to na z czasem poprawiał. Zwłaszcza że zmieni nam się zbiór danych i wyleci jedna kolumna, więc i tak będzie to do poprawy.

Reply:
@JakubDeka co z tym PR?

Reply:
dzisiaj wchodzi uaktualnienie PR